routes/snow.py

```diff
@@ -0,0 +1,201 @@
+    get_dim_encodings,
+    deepflatten,
+    build_csv_dicts,
+    write_csv,
+)
+from validate_request import (
+    validate_latlon,
+    project_latlon,
+)
+from validate_data import nullify_and_prune, postprocess
+from . import routes
+from config import WEST_BBOX, EAST_BBOX
+
+snow_api = Blueprint("snow_api", __name__)
+# rasdaman targets
+sfe_coverage_id = "mean_annual_snowfall"
+
+
+def package_sfe_data(sfe_resp):
+    """Package the SFE data into a nested JSON-like dict.
+
+    Arguments:
+        sfe_resp -- the response(s) from the WCS GetCoverage request(s).
+
+    Returns:
+        di -- a nested dictionary of all SFE values
+    """
+    # intialize the output dict
+    sfe_encodings = asyncio.run(get_dim_encodings(sfe_coverage_id))
+    models = list(sfe_encodings["model"].values())
+    scenarios = list(sfe_encodings["scenario"].values())
+    decades = list(sfe_encodings["decade"].values())
+    di = {
+        m: {sc: {dec: {"SFE": None} for dec in decades} for sc in scenarios}
+        for m in models
+    }
+    # populate the dict with the response
+    flat_list = list(deepflatten(sfe_resp))
+    i = 0
+    for model in di.keys():
+        for scenario in di[model].keys():
+            for decade in di[model][scenario]:
+                di[model][scenario][decade]["SFE"] = flat_list[i]
+                i += 1
+    # remove the nonsense encoding combinations
+    # e.g., CRU-TS/RCP8.5 or NCAR-CCSM4/1920-1929
+    projection_decades = [
+        "2010-2019",
+        "2020-2029",
+        "2030-2039",
+        "2040-2049",
+        "2050-2059",
+        "2060-2069",
+        "2070-2079",
+        "2080-2089",
+        "2090-2099",
+    ]
+    historical_decades = list(set(decades) - set(projection_decades))
+    di["CRU-TS"].pop("rcp45", None)
+    di["CRU-TS"].pop("rcp60", None)
+    di["CRU-TS"].pop("rcp85", None)
+    for proj_dec in projection_decades:
+        di["CRU-TS"]["historical"].pop(proj_dec, None)
+    for model in ["GFDL-CM3", "GISS-E2-R", "IPSL-CM5A-LR", "MRI-CGCM3", "NCAR-CCSM4"]:
+        di[model].pop("historical")
+        for scenario in ["rcp45", "rcp60", "rcp85"]:
+            for hist_dec in historical_decades:
+                di[model][scenario].pop(hist_dec, None)
+    return di
+
+
+def summarize_mmm_sfe(all_sfe_di):
+    """Generate min-mean-max summaries of the historical and projected SFE data.
+
+    Arguments:
+        all_sfe_di -- the intial nested dict package of all SFE data
+
+    Returns:
+        mmm_sfe_di -- a nested dict that is a subset of the intial package
+    """
+    mmm_sfe_di = {}
+    mmm_sfe_di["historical"] = {}
+    mmm_sfe_di["projected"] = {}
+    hist_vals = [
+        all_sfe_di["CRU-TS"]["historical"][k]["SFE"]
+        for k in all_sfe_di["CRU-TS"]["historical"].keys()
+    ]
+    mmm_sfe_di["historical"]["sfemin"] = min(hist_vals)
+    mmm_sfe_di["historical"]["sfemax"] = max(hist_vals)
+    mmm_sfe_di["historical"]["sfemean"] = int(np.mean(hist_vals))
```

Comment:
Having a digit in the tenths place for inches of water makes sense to me! So before we merge this should I go ahead and re-process and re-ingest the snowfall coverage as described (XREF https://github.com/ua-snap/rasdaman-ingest/issues/34) so we can get one decimal place over here in the API, or should I "fake it" for now for the min and max (e.g, just add `.0` on the end so that a min of `4` becomes `4.0`) and then compute the mean with one decimal?

Reply:
I think just changing lines 101 and 113 to use the `round` function instead of `int` would be sufficient. The min/max values should already be good to go for when the underlying values are updated I think!